content/2020-06-19-how-to-learn-typescript-quickly/index.md

```diff
@@ -0,0 +1,259 @@
+
+## Make your Map
+
+The first step is to make your map. This will help answer questions like:
+
+- What does it mean to "learn TypeScript"?
+- How long will the project take?
+- What are the expected outcomes?
+- How am I going to organize my learning project?
+
+### Write a Clear Vision
+
+For the vision, define your project and use the Objective Key Results (OKRs) framework.  If you aren't familiar, this was popularized by the book "[Measure What Matters](https://www.whatmatters.com/)" and is used at companies like Google and Intuit.
+
+First, define what you expect to achieve by learning TypeScript.
+
+In my learning project, I wrote:
+
+> Learn TypeScript deeply and become the [Company] "in-house expert"
+
+That will be your **Objective**. Next, figure out how you're going to *measure* it. We refer to these as the **Key Results**. Continuing with my previous project, mine were:
+
+- Read "Programming TypeScript" by Boris Cherny
+
+- Read the [Official TypeScript Handbook](https://www.typescriptlang.org/docs/handbook/basic-types.html)
+
+- Produce small code examples for all concepts
+
+- Contribute to TypeScript, [the language](https://github.com/microsoft/TypeScript/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue")
+- Contribute to [typescript-cheatsheets/react-typescript-cheatsheet](https://github.com/typescript-cheatsheets/react-typescript-cheatsheet)
+
+**Don't rush through this part**. Spend some time thinking about your objective and what you need to achieve to get there. Completing your key results should allow you to confidently say you've completed your objective.
+
+With this vision in mind, you now have clarity on it means when you say "learn TypeScript." and you know what outcomes you expect to achieve.
+
+### Establish a Timeframe
+
+You don't want to be racing on this map forever. You need to be realistic and ask yourself how much time we have for this. Estimating time to complete tasks in software is hard. We'll try our best by using something I like to call the **Timeframe Formula**:
+
+Take the length of months you want to work on this, multiple by 4 to give the amount in weeks. Then multiple it by the number of hours per week that you want to dedicate.
+
+https://excalidraw.com/#json=4877101601652736,SeRw5x2aEzwtztaHHk0xUQ
+
+For example, when I did a similar project, I gave myself 3 months with 8 hours per week. Using the Timeframe Formula, you get:
+
+(3 months * 4 weeks/month) * 8 hours/week = 96 hours total.
+
+https://excalidraw.com/#json=5134809571524608,ASGbR83wxxoafOuInmZ5Ew
+
+You now have your timeframe and expectation of how much you plan to dedicate each week.
+
+### Build a Timeline
+
+The next thing we need to add to our map is a timeline. Think of this as a week by week calendar. It helps you think less because you have already decided what to do for that week. It won't be perfect. You may need to readjust, but at least you will have established the framing.
```

Comment:
```suggestion
The next thing you need to add to you map is a timeline. Think of this as a week by week calendar. It helps you think less because you will already know what to do for that week. It won't be perfect. You may need to readjust, but you will have removed friction.
```